README.md

```diff
@@ -65,66 +125,157 @@ Usage: `data-diff DB1_URI TABLE1_NAME DB2_URI TABLE2_NAME [OPTIONS]`
 Options:
 
   - `--help` - Show help message and exit.
-  - `-k` or `--key_column` - Name of the primary key column
+  - `-k` or `--key-column` - Name of the primary key column
   - `-t` or `--update-column` - Name of updated_at/last_updated column
   - `-c` or `--columns` - List of names of extra columns to compare
   - `-l` or `--limit` - Maximum number of differences to find (limits maximum bandwidth and runtime)
   - `-s` or `--stats` - Print stats instead of a detailed diff
   - `-d` or `--debug` - Print debug info
   - `-v` or `--verbose` - Print extra info
   - `-i` or `--interactive` - Confirm queries, implies `--debug`
+  - `--max-age` - Considers only rows younger than specified. See `--min-age`.
+                  Useful for specifying replication lag.
   - `--min-age` - Considers only rows older than specified.
-                  Example: `--min-age=5min` ignores rows from the last 5 minutes.
+                  Example: `--min-age=5min` considers only rows from the last 5 minutes.
```

Comment:
It's ignore, not considers.

It looks like it's really confusing, so maybe we should re-think the name or phrasing?

Reply:
Hmmmm yeah we definitely need to rethink it, but isn't this right?

Drew a diagram below... Here the `--min-age` is the `id >= now() -5min` and `--max-age` is `id <= now() - 2min`?

```
┌ ─ ─ ┬─────┬─────┬─────┬─────┬ ─ ─ ┬ ─ ─ ┬ ─ ─ ┐
 11:59│12:00│12:01│12:02│12:03│12:04 12:05 12:06
└ ─ ─ ┴──▲──┴─────┴─────┴──▲──┴ ─ ─ ┴ ─ ▲ ┴ ─ ─ ┘
         │                 │            │
         │                 │            │
         │                 │            │
  ┌──────┴───────┐    ┌────┴─────────┐  │ ┌─────────────┐
  │ --min-age=5m │    │ --max-age=2m │  └─┤Present time │
  └──────────────┘    └──────────────┘    └─────────────┘
```

Reply:
No, min-age at this time means anything before 12.

And a min-age that's bigger than the max-age shouldn't be allowed.

When you go back in time the "time value" or timestamp get smaller, that's true, but the age actually gets bigger. (like someone born in the 80s is "bigger" than someone from the 90s, despite the number being smaller)

The "age" of 12:00 is 5 mins. The age of 12:01 is 4 mins, which is below 5, aka the minimum age, so it's out.

The "age" of 11:00 is an hour and 5 mins, which is a bigger age than the minimum age.

Does it make sense?

Reply:
Yeah, but I mean... I drew it, and I'm still confused 😓  

But I think we need to change it to `--min-updated-key` and `--max-updated-key` which would line up with the drawing above

Reply:
But then it's not very clear that the value is relative to now. It sounds more like an absolute value..

Reply:
Yeah, but I think it's clearer then if you do `--min-updated -5min` or something.

Reply:
Hi Y'all, end user here.  I think the semantics of “age” matter a lot.  The source of confusion is that “age” is implicitly negative, but we think about it as a positive number/absolute value, and so inverts the conventional meaning of “min” and “max” when thinking about timestamps.  Your readme documents `--min-age` being a lower bound on “age” and an upper bound on “timestamp” (matching my intuition), but the implementation does the reverse.

Regardless, docs should match readme, and ideally should match user intuition as well.